File: app/src/main/java/mega/privacy/android/app/lollipop/megaachievements/ReferralBonusesFragment.java
package mega.privacy.android.app.lollipop.megaachievements;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.RelativeLayout;

import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import mega.privacy.android.app.R;
import mega.privacy.android.app.components.SimpleDividerItemDecoration;
import mega.privacy.android.app.fragments.BaseFragment;
import mega.privacy.android.app.listeners.GetAchievementsListener;

import static mega.privacy.android.app.lollipop.megaachievements.AchievementsActivity.sFetcher;
import static mega.privacy.android.app.utils.LogUtil.logDebug;

public class ReferralBonusesFragment extends BaseFragment implements OnClickListener
		, GetAchievementsListener.DataCallback{
	RelativeLayout parentRelativeLayout;
	RecyclerView recyclerView;
	LinearLayoutManager mLayoutManager;
	MegaReferralBonusesAdapter adapter;
	
	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
		logDebug("onCreateView");

		boolean enabledAchievements = megaApi.isAchievementsEnabled();
		logDebug("The achievements are: " + enabledAchievements);

		View v = inflater.inflate(R.layout.fragment_referral_bonuses, container, false);

		parentRelativeLayout = (RelativeLayout) v.findViewById(R.id.referral_bonuses_relative_layout);

		recyclerView = (RecyclerView) v.findViewById(R.id.referral_bonuses_recycler_view);
		recyclerView.addItemDecoration(new SimpleDividerItemDecoration(context));
		mLayoutManager = new LinearLayoutManager(context);
		recyclerView.setLayoutManager(mLayoutManager);
		recyclerView.setItemAnimator(new DefaultItemAnimator());

		return v;
	}

	@Override
	public void onActivityCreated(@Nullable Bundle savedInstanceState) {
		super.onActivityCreated(savedInstanceState);

		// Activity actionbar has been created which might be accessed by UpdateUI().
		if (mActivity != null) {
			ActionBar actionBar = ((AppCompatActivity) mActivity).getSupportActionBar();
			if (actionBar != null) {
				actionBar.setTitle(getString(R.string.title_referral_bonuses));
			}
		}

		// The root view has been created, fill it with the data when data ready
		if (sFetcher != null) {
			sFetcher.setDataCallback(this);
		}
	}

	@Override
	public void onClick(View v) {
		logDebug("onClick");
		switch (v.getId()) {
			case R.id.referral_bonuses_layout:{
				logDebug("Go to section Referral bonuses");
				break;
			}
		}
	}

	public int onBackPressed(){
		logDebug("onBackPressed");
		return 0;
	}

	private void updateUI() {
		if (context == null || sFetcher == null) return;

		ArrayList<ReferralBonus> bonuses = sFetcher.getReferralBonuses();
		if (bonuses.size() == 0) return;

		if (adapter == null) {
			adapter = new MegaReferralBonusesAdapter(context, this, bonuses, recyclerView);
		} else {
			adapter.setReferralBonuses(bonuses);
		}

		recyclerView.setAdapter(adapter);
	}

	@Override
	public void onAchievementsReceived() {
		updateUI();
	}
}

File: app/src/main/java/mega/privacy/android/app/presentation/achievements/referral/ReferralBonusesFragment.kt
package mega.privacy.android.app.presentation.achievements.referral

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.runtime.getValue
import androidx.compose.ui.platform.ComposeView
import androidx.compose.ui.platform.ViewCompositionStrategy
import androidx.fragment.app.Fragment
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.CoroutineDispatcher
import mega.privacy.android.app.R
import mega.privacy.android.app.presentation.achievements.referral.view.ReferralBonusRoute
import mega.privacy.android.app.presentation.extensions.isDarkMode
import mega.privacy.android.core.ui.theme.AndroidTheme
import mega.privacy.android.data.qualifier.MegaApi
import mega.privacy.android.domain.entity.ThemeMode
import mega.privacy.android.domain.qualifier.IoDispatcher
import mega.privacy.android.domain.usecase.GetThemeMode
import nz.mega.sdk.MegaApiAndroid
import javax.inject.Inject

/**
 * ReferralBonusesFragment
 */
@AndroidEntryPoint
class ReferralBonusesFragment : Fragment() {
    /**
     * [MegaApiAndroid] injection
     */
    @Inject
    @MegaApi
    lateinit var megaApi: MegaApiAndroid

    /**
     * ioDispatcher as [CoroutineDispatcher] injection
     */
    @Inject
    @IoDispatcher
    lateinit var ioDispatcher: CoroutineDispatcher

    /**
     * Get system's default theme mode
     */
    @Inject
    lateinit var getThemeMode: GetThemeMode

    /**
     * onCreateView
     */
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View = ComposeView(requireContext()).apply {
        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
        setContent {
            val themeMode by getThemeMode()
                .collectAsStateWithLifecycle(initialValue = ThemeMode.System)

            AndroidTheme(isDark = themeMode.isDarkMode()) {
                ReferralBonusRoute()
            }
        }
    }

    /**
     * onViewCreated
     */
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        // Activity actionbar has been created which might be accessed by UpdateUI().
        (requireActivity() as? AppCompatActivity)?.supportActionBar?.title =
            getString(R.string.title_referral_bonuses)
    }
}